rotkehlchen/db/history_events.py

```diff
@@ -363,11 +362,10 @@ def get_evm_event_by_identifier(self, identifier: int) -> Optional['EvmEvent']:
 
         return deserialized
 
+    @staticmethod
     def _create_history_events_query(
-            self,
             filter_query: HistoryBaseEntryFilterQuery,
-            entries_limit: int,
-            has_premium: bool,
+            entries_limit: int | None,
```

Comment:
so as I wrote elsewhere this is a terrible hack with how you use it in the codebase/tests. Just create separate functions.